### PR TITLE
map-hover-interaction

### PR DESCRIPTION
**Pull request by m-kavita**
updates on hover for countries and map style + countries names on click appear in console log

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -9,15 +9,15 @@ mapboxgl.accessToken = 'pk.eyJ1Ijoia2F2aXRhLW0iLCJhIjoiY2t6eHI1MDE5MDFxZzJ5czBkc
...
+  const [lng, setLng] = useState(13.0138);
+  const [lat, setLat] = useState(26.9115);
+  const [zoom, setZoom] = useState(1.66);
 
   useEffect(() => {
   if (map.current) return; // initialize map only once
     map.current = new mapboxgl.Map({
       container: mapContainer.current,
-      style: 'mapbox://styles/mapbox/streets-v11',
+      style: 'mapbox://styles/kavita-m/cl08ayfiw000c15o3e9q8ur3f',
```
Comment (omguhh) [inline src/App.js:20]: why are you using a custom style? 🤔

File: src/App.js [hunk 2]
```diff
@@ -32,6 +32,81 @@ function App() {
...
+      "id": "cfh",  // country-fills-hover",
+      "type": "fill",
+      "source": "cbs",
+      "layout": {},
+      "paint": {
+          "fill-color": "#BFBAAD",
+          "fill-opacity": 0.5
+      },
+      "filter": ["==", "name", ""]
+    });
```
Comment (omguhh) [inline src/App.js]: can we move all this into a seperate function and call it initMap ? 

```suggestion
function initMap () => {
    map.current.on('load', () => {
    map.current.addSource('cbs', {
    'type': 'geojson',
    'data': 'https://d2ad6b4ur7yvpq.cloudfront.net/naturalearth-3.3.0/ne_50m_admin_0_countries.geojson'
    });

    map.current.addLayer({
      'id': 'cf',         // country-fills
      'type': 'fill',
      'source': 'cbs',
      'layout': {},
      'paint': {
      'fill-color': '#BFBAAD',
      'fill-opacity': 0.1
      }
    });
    
    map.current.addLayer({
    'id': 'cb',         // country-borders
    'type': 'line',
    'source': 'cbs',
    'layout': {},
    'paint': {
    'line-color': '#BFBAAD',
    'line-width': 1
    }
    });

    map.current.addLayer({
      "id": "cfh",  // country-fills-hover",
      "type": "fill",
      "source": "cbs",
      "layout": {},
      "paint": {
          "fill-color": "#BFBAAD",
          "fill-opacity": 0.5
      },
      "filter": ["==", "name", ""]
    });
 }
 
 useEffect(() => 
 if (!map.current) return; 
 this.initMap()
 
 )
```

File: src/App.js [hunk 3]
```diff
@@ -9,15 +9,15 @@ mapboxgl.accessToken = 'pk.eyJ1Ijoia2F2aXRhLW0iLCJhIjoiY2t6eHI1MDE5MDFxZzJ5czBkc
 function App() {
```
Comment (omguhh) [inline src/App.js:9]: you could create a Map component and then import it into App(), so app stays clean 🤔

**Comment by omguhh** [comment]
Awesome work 🚀   I checked the preview and it works well. I've left some code style comments :D

File: src/App.js (hunk 1 above)
Comment (m-kavita) [reply src/App.js:20]: makes the map look nicer!

File: src/App.js [hunk 4]
```diff
@@ -74,6 +75,13 @@ function App() {
       },
       "filter": ["==", "name", ""]
     });
+ }
+ 
+ useEffect(() => 
+ if (!map.current) return; 
+ this.initMap()
```
Comment (omguhh) [inline src/App.js]: ⭐

File: src/App.js [hunk 5]
```diff
@@ -32,6 +32,89 @@ function App() {
...
+      "id": "cfh",  // country-fills-hover",
+      "type": "fill",
+      "source": "cbs",
+      "layout": {},
+      "paint": {
+          "fill-color": "#BFBAAD",
+          "fill-opacity": 0.5
+      },
+      "filter": ["==", "name", ""]
+    });
```
Comment (omguhh) [inline src/App.js]: you can take this out completely of useEffect()

**Review by omguhh — APPROVED** [review]

**Pull request merged**